File: app/src/main/java/software/credible/abercrombiefitchkata/remote/PromotionsApi.java
package software.credible.abercrombiefitchkata.remote;

import io.reactivex.Single;
import retrofit2.http.GET;
import software.credible.abercrombiefitchkata.dto.PromotionsResponseDto;

public interface PromotionsApi {

    @GET("promotions.json")
    Single<PromotionsResponseDto> getPromotions();

}

File: app/src/main/java/software/credible/abercrombiefitchkata/remote/PromotionsApi.kt
package software.credible.abercrombiefitchkata.remote

import io.reactivex.Single
import retrofit2.http.GET
import software.credible.abercrombiefitchkata.dto.PromotionsResponseDto

interface PromotionsApi {

    @get:GET("promotions.json")
    val promotions: Single<PromotionsResponseDto>

}
